### PR TITLE
Tune-up navigation

### PR DESCRIPTION
**Pull request by devinivy**

### CODE REVIEW

**Comment by devinivy** [comment]
@jdonisvitch mind reviewing this and if it looks good merging?  It will waterfall into #8.

File: lib/public/js/docs-nav.js [hunk 1]
```diff
@@ -1,41 +1,140 @@
...
-            const item = internals.render(
-                internals.item(text, el.tagName.replace('H', 'indent-'))
+            return internals.render(
+                internals.item(el.innerHTML, anchor.hash, indent)
             );
+        },
+        onScroll: (navItem) => {
 
-            item.addEventListener('click', () => el.scrollIntoView({ block: 'start', behavior: 'smooth' }));
+            if (window.location.hash !== navItem.hash && !hashState.autoScrolling && !hashState.firstScroll) {
```
Comment (jdonisvitch) [inline lib/public/js/docs-nav.js]: fun stuff ;)

Comment (devinivy) [reply lib/public/js/docs-nav.js]: Hahaha, ahhhh I can't wait to clean this up!

**Comment by devinivy** [comment]
Thanks for the review, pal!  Merge into api-docs-fe-mobile / #8 whenever you think is 👍 👍

**Pull request merged**